core/include/sched.h

```diff
@@ -188,6 +188,25 @@ void sched_register_cb(void (*callback)(uint32_t, uint32_t));
 
 #endif
 
+#ifdef BOARD_NATIVE
```

Comment:
I would prefer to include a native header here that includes the following code block instead of having it directly part of `sched.h`.